pretyping/evarconv.ml

```diff
@@ -176,6 +176,10 @@ let check_conv_record env sigma (t1,sk1) (t2,sk2) =
         let s = ESorts.kind sigma s in
 	lookup_canonical_conversion
 	  (proji, Sort_cs (family_of_sort s)),[]
+      | Proj (p, c) ->
+        let c2 = Globnames.ConstRef (Projection.constant p) in
+        let sk2 = Stack.append_app [|c|] sk2 in
+        lookup_canonical_conversion (proji, Const_cs c2), sk2
       | _ ->
```

Comment:
Warning 4 does that. It is disabled because it is triggered by 2855 lines (https://gist.github.com/SkySkimmer/4591058185a10c6f09016afd66d978d1).
That isn't to say it couldn't be turned on, I was trying to be conservative in #582.